### ISSUE TITLE
[Files Module] Folder creation forces camel case for folder name, should allow dashes

### ISSUE THREAD

**Issue by mattcdavis1** [pyrocms/pyrocms#4241]
This seems to be a front end javascript issue only (forcing underscores).  I modified the folder name via javascript (changed underscore to dash) and it passed backend validation.

**Comment by mattcdavis1** [comment]
ok - looks like even though validation doesn't fail backend still forces underscores

**Comment by RyanThompson** [comment]
Folders are used to determine meta entry table names so they need to be kept underscore.

**Comment by mattcdavis1** [comment]
@RyanThompson - these limitations are frustrating (and in this case needless).  Why not just translate the table name to underscore and still allow flexibility for the folder to be dashed?  Or just use the folder id in the table name.

**Comment by RyanThompson** [comment]
Can I ask what you are doing that this is causing such frustrating issues for? Maybe if I understand where you're coming from better I can discuss this with a helpful context to what you are trying to do.

But "limitations" has me worried about what you're doing with it that you NEED dashes. The only thing I can think of is presentation to public content but in that case why don't you convert the systems underscores to dashes?

If it's something like S3 buckets.. you could also easily maintain your own fork (others do) with whatever naming / conventions you see fit too.

I prefer to keep core simple, consistent, and clean to itself. And allow developers to extend / override as needed.

Hopefully this is such a case where we can get you what you need!

**Comment by mattcdavis1** [comment]
The frustration for me stems from a lack of control more than anything.   It's compounded in this case by the fact that i'm already limited to one folder level - so i've already had to deal with a major limitation and then i turn around and have to deal with another setback.  

To some extent it's a principle thing - I don't like my software making decisions for me.  Also in part, is the fact that i'm not accustomed to these limitations with Craft.  And then there's the fact that i simply prefer dashes in URLs.  And lastly there's the fact that it's just simply not necessary.  I know you mentioned the technical reason but that is easily worked around.

Imagine in another context, say a stream entry, if there were arbitrary limitations like - you can't create an entry with a slug that starts with "w" or ends with "k".  Well, you could probably work around that fairly easily but it would be frustrating right?  There's no technical need for this limitation and yet it is being imposed on you nonetheless.

In summary - it's more of an issue of control (of URLs in this case), preference, prior experience  (of having total freedom over my asset URLs) and principle rather than "need".
